### ISSUE TITLE
Erractic location

### ISSUE THREAD

**Issue by collectorgeneral** [y20k/trackbook#4]
Beta apk from #2 ,

Recorded a bunch of tracks today. Noticed on one of my runs, the distance was way off. It said about 5KM, though I  just walked 2. I noticed that some of the points were completely erratic. One was on the other end of the road, which might explain the distance. 
I suspect this is due to the location from network rather than GPS, since I was inside a store when I noticed this. Is it possible to have an option where we can select only GPS instead of GPS+Network?

**Comment by y20k** [comment]
Hi there. 

Thanks for finding this. I noticed that kind of behavior in earlier versions of Trackbook, too. One known problem is: If you stop at a location where there is no GPS Trackbook will eventually try to use a Network location fix. Those locations tend to be inaccurate. And if you stay for a while several locations will pop up left and right of your "real" location and land on the map.

It is not easy to differentiate between walking around within a large building (no GPS) and just sitting in a building. Those erratic markers can sometimes be interpreted as movement.

I will have a second look on the the part of code that tries to decide that.

**Comment by collectorgeneral** [comment]
Yes I agree that it is really hard to differentiate such signal.  I am thinking you could use 3-4  readings and then pick the most likely one out of them. So if 3 out of 4 points are close to location 'A' and one  point is 2 KM away, I know that the 2KM away was likely a glitch. Something like that.

I thought maybe you could have an option that only uses GPS points. If it does not acquire points, then it will wait until there is a lock on again and then continue as usual. I think this might be easier?

**Comment by y20k** [comment]
I just wanted to let you know that I have an idea on how to solve at least a special case of the "Erractic location" bug. I think Trackbook can pause searching for locations if the device is lying flat on the table somewhere. I can use the Accelerometer to detect non-movement. It is not implemented yet, but will be in a future release. In the scenario "recording tracks, but having a coffee in a location without a GPS fix (eg. a mall)" Trackbook would pause the recoding as long the device is not being moved. I know: That is just one aspect of the problem ... but: one step at a time.

**Comment by collectorgeneral** [comment]
@y20k  Yes of course, one step at a time. 

Seems like a good idea. I would suggest have a buffer time, say like no movement for 30 seconds? Then you could stall the recording until you receive the information that the device has moved again.
